### PR TITLE
Refactor consents routes, nest patients page

### PR DESCRIPTION
**Pull request by misaka**
This rejigs the routes so that we can nest the patients page under the
correct consents tab:

/sessions/:session_id/consents/:tab/patients/:patient_id

And also as a side-effect changes the generated helpers for the routes,
to something more sensible imo:

consents_session_path -> session_consents_path

### CODE REVIEW

File: app/components/app_consent_component.rb [hunk 1]
```diff
@@ -43,11 +44,10 @@ def contact_parent_or_guardian_link(consents)
     link_to(
       "Contact #{consent.parent_name} (the #{role} who refused)",
       session_patient_manage_consent_path(
-        session,
-        patient,
-        @route,
-        consent.id,
-        :agree
+        session_id: session,
```
Comment (tvararu) [inline app/components/app_consent_component.rb:47]: Hmm, does this work as-is? Maybe for consistency:

```suggestion
        session_id: session.id,
```

Or if the var was an id all along:

```suggestion
        session_id:,
```

File: app/controllers/manage_consents_controller.rb [hunk 2]
```diff
@@ -83,16 +83,16 @@ def current_step
...
-    case @route
+    case @section
     when "consents"
-      consents_session_path(@session)
+      session_consents_path(@session)
     when "triage"
       triage_session_path(@session)
     else
-      vaccinations_session_path(@session)
+      session_vaccinations_path(@session)
```
Comment (tvararu) [inline app/controllers/manage_consents_controller.rb:95]: - session_consents_path
- triage_session_path
- session_vaccinations_path

Triage is the odd one out, how come? I thought it should be `session_triage_path` according to the new routes

File: app/controllers/patient_sessions_controller.rb [hunk 3]
```diff
@@ -47,11 +47,11 @@ def set_back_link
     @back_link =
       case @route
       when "vaccinations"
-        vaccinations_session_path(@session)
+        session_vaccinations_path(@session)
       when "triage"
         triage_session_path(@session)
       when "consents"
-        consents_session_path(@session)
+        session_consents_path(@session)
```
Comment (tvararu) [inline app/controllers/patient_sessions_controller.rb:54]: See above

File: app/controllers/vaccinations_controller.rb [hunk 4]
```diff
@@ -90,34 +91,37 @@ def create
         if @draft_vaccination_record.delivery_site_other
           redirect_to edit_session_patient_vaccinations_delivery_site_path(
                         @session,
-                        @patient
+                        patient_id: @patient
```
Comment (tvararu) [inline app/controllers/vaccinations_controller.rb:94]:
```suggestion
                        patient_id: @patient.id
```

File: app/controllers/vaccinations_controller.rb [hunk 5]
```diff
@@ -90,34 +91,37 @@ def create
...
                       )
         elsif @draft_vaccination_record.batch_id.present?
           redirect_to confirm_session_patient_vaccinations_path(
                         @session,
-                        @patient
+                        patient_id: @patient
```
Comment (tvararu) [inline app/controllers/vaccinations_controller.rb:99]:
```suggestion
                        patient_id: @patient.id
```

File: app/controllers/vaccinations_controller.rb [hunk 6]
```diff
@@ -90,34 +91,37 @@ def create
...
                         @session,
-                        @patient
+                        patient_id: @patient.id
                       )
         end
       else
         redirect_to edit_reason_session_patient_vaccinations_path(
                       @session,
-                      @patient
+                      patient_id: @patient
```
Comment (tvararu) [inline app/controllers/vaccinations_controller.rb:110]:
```suggestion
                      patient_id: @patient.id
```

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -30,15 +30,20 @@ def page_title(service_name)
...
+  # end
+
+  # def session_patient_vaccinations_path(*args, **kwargs)
+  #   session_patient_path(
+  #     *args,
+  #     **kwargs,
+  #     section: "vaccinations",
+  #     tab: params[:tab]
+  #   )
+  # end
```
Comment (tvararu) [inline app/helpers/application_helper.rb:48]:
```suggestion
```

File: app/views/sessions/show.html.erb [hunk 8]
```diff
@@ -32,16 +34,22 @@
         </p>
 
         <h2 class="nhsuk-heading-s nhsuk-u-margin-bottom-1">
-          <%= link_to "Triage health questions",
-            triage_session_path(@session) %>
+          <%= link_to(
+            "Triage health questions",
+            # TODO: Can we get rid of the section param here?
```
Comment (tvararu) [inline app/views/sessions/show.html.erb:39]: Worth checking this in like a timeboxed ~5 minutes but I'd remove these TODOs before merging even if we don't. It's really not that bad having to specify it just in this instance.

File: config/initializers/tab_paths.rb [hunk 9]
```diff
@@ -19,3 +19,5 @@
     "could-not" => :could_not_vaccinate
   }
 }.with_indifferent_access
+
+# TAB_SECTIONS = TAB_PATHS.keys
```
Comment (tvararu) [inline config/initializers/tab_paths.rb:23]:
```suggestion
```

**Review by tvararu — APPROVED** [review]
New routes and URLs are super clean 👍 

New path helpers also clearer

Back link works predictably based on a bit of testing

Left some comments but pre-approving pending commit tidy up, well done! 🙌

File: app/components/app_consent_component.rb (hunk 1 above)
Comment (misaka) [reply app/components/app_consent_component.rb:47]: In my experimentation, `session_id` can be set to either `session` or `session.id`, so the original code above should work, ya. However, like you suggest, I've been trying to standardise on `session.id`, I probably just missed that one. It's a shame that `session: session` doesn't work as that could've been nicely abbreviated as `session:`

**Comment by misaka** [comment]
The comments are spot-on, and things that just haven't been changed because this is still a WIP. I'll update it, probably addressing most if not all the comments here, and then maybe you can have another look @tvararu . Cheers!

**Pull request merged**